Automattic/wp-calypso
# Map Block, With Asynchronous Loader

#### Changes proposed in this Pull Request

This is identical to the [Map block PR](https://github.com/Automattic/wp-calypso/pull/27836) except that it replaces the [Calypso loadScript utility](https://github.com/Automattic/wp-calypso/tree/0964c39fb81390eb681074037bb001510fd1cb0c/client/gutenberg/extensions/map/load-script) with a custom async loader utility that manages scripts and stylesheets. In the Map block, this utility is used to load the Mapbox [script](https://api.tiles.mapbox.com/mapbox-gl-js/v0.51.0/mapbox-gl.js) and [stylesheet](https://api.tiles.mapbox.com/mapbox-gl-js/v0.51.0/mapbox-gl.css) at runtime. At some point in the future, making a library like this available in Gutenberg core might be a valuable addition for plugin authors (cc: @mtias). 

Also, Map block has been removed from the Jetpack preset and added to Jetpack Beta, so `JETPACK_BETA_BLOCKS` constant will need to be set to test.

#### Testing instructions

Jurassic Ninja link: https://jurassic.ninja/create/?gutenberg&gutenpack&shortlived&jetpack-beta&calypsobranch=try/map+loader&branch=try/google-maps-api-key

- Click `Set Up Jetpack`
- Click `Start with free`
- Click `Activate recommended features`
- ~~SSH in using connection string and password found in the persistent admin notice in the site’s wp-admin~~
- ~~`cd ~/apps/$USER/public`~~
- ~~`wp config set JETPACK_BETA_BLOCKS true --raw --type=constant`~~
- Navigate to `Settings->Jetpack Constants`
- Check `JETPACK_BETA_BLOCKS` and Save.
- Create new Post, add block, search for `Map`
- Follow instructions to get Mapbox key
